appsmithorg/appsmith
# feat: Added file id mapping in datasource config

## Description

This PR adds:
- File Id mapping of the google sheets selected by user, in datasource configuration, so that when creating queries on top of such gsheet datasource, only the selected spreadsheets can be seen in the spreadsheet dropdown.

Changes done on client side:
- As soon as user selects file in file picker popup, the callback will get the file ids and update the datasource to contain file ids as a part of datasource configuration properties.
- If user cancels the file selection, file ids is sent as empty array and datasource is updated.

Changes done on server side:
- In `GoogleSheetPlugin.java` where we have defined execute and trigger methods for gsheet query, here I have added a new variable allowedFileIds, which gets the list of authorised file ids from datasource configuration object and the same list is passed to functions like `transformTriggerResponse` and `transformExecutionResponse`, which returns  file list data based on the allowedFileIds. In FileListMethod class, these methods contain the logic to send only authorised file data.
- Since these two methods are a part of triggerMethod and executionMethod interfaces, all gsheet query operation classes that extend this method, their function definition needed to be updated with this third allowedFileIds parameter.
- Similarly all gsheet query operations test classes were using these two methods, and hence this third parameter needed to be added there as well.

How to test:
- With this improvement, when we select `file1` and `file2` for one datasource and `file3` and `file4` for another datasource, In the query dropdown for first ds, we should only see `file1` and `file2`, for second datasource, we should only see `file3` and `file4`.
- Please check following gsheet operations:
    - Fetch Many
    - Fetch Details
    - Update One
    - Update Many
    - Insert Many

> Add a TL;DR when description is extra long (helps content team)

Fixes #21074


Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video


## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)


## How Has This Been Tested?

- Manual
- JUnit

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity


### QA activity:
